compiler/rustc_errors/src/emitter.rs

```diff
@@ -2709,23 +2709,48 @@ impl Style {
     }
 }
 
-/// Whether the original and suggested code are visually similar enough to warrant extra wording.
+/// Determines if the original code and the suggested code have sufficient visual similarity 
+/// to include extra textual information or descriptions to highlight the similarities or 
+/// differences between them. 
+/// 
+/// # Returns
+///
+/// Returns `true` if the original and suggested code are visually similar enough to warrant 
+/// extra wording, otherwise `false`.
 pub fn is_case_difference(sm: &SourceMap, suggested: &str, sp: Span) -> bool {
     // FIXME: this should probably be extended to also account for `FO0` → `FOO` and unicode.
+    // Retrieve code
     let found = match sm.span_to_snippet(sp) {
-        Ok(snippet) => snippet,
+        Ok(snippet) => {
+            snippet
+        }
         Err(e) => {
             warn!(error = ?e, "Invalid span {:?}", sp);
             return false;
         }
     };
-    let ascii_confusables = &['c', 'f', 'i', 'k', 'o', 's', 'u', 'v', 'w', 'x', 'y', 'z'];
-    // All the chars that differ in capitalization are confusable (above):
-    let confusable = iter::zip(found.chars(), suggested.chars())
-        .filter(|(f, s)| f != s)
-        .all(|(f, s)| (ascii_confusables.contains(&f) || ascii_confusables.contains(&s)));
-    confusable && found.to_lowercase() == suggested.to_lowercase()
-            // FIXME: We sometimes suggest the same thing we already have, which is a
-            //        bug, but be defensive against that here.
-            && found != suggested
+
+    // ASCII confusable characters when capitalization.
+    const ASCII_CONFUSABLES: [char; 12] = ['c', 'f', 'i', 'k', 'o', 's', 'u', 'v', 'w', 'x', 'y', 'z'];
```

Comment:
Using a const does not make a difference, actually. Consts are copied on access, so this is actually worse, if anything (as we now potentially spill two copies on the attach, instead of one).

Not that it actually matters, given than this is the error path, but still. You should probably change the type to `&[char; 12]` if you want to make it const (do that only the reference is copied, not the array).